### ISSUE TITLE
 ufunc 'correct_alternative_cosine' did not contain a loop with signature matching types

### ISSUE THREAD

**Issue by garchaaman19** [lmcinnes/umap#1040]
`import numpy as np
import pandas as pd
import umap
import hdbscan

 embeddings = my_embedder.encode(
                    data_df.normalization.values, show_progress_bar=False)
       
                umap_embeddings = umap.UMAP(n_neighbors=np.min([5, data_df.shape[0]]),
                                            n_components=3,
                                            metric='cosine',
                                            random_state=17
                                            ).fit_transform(embeddings)
`

This is only failing for rows > 5k, It works for other set of data <2.5K and if I split the 5k file into two parts then it works fine. 
I cannot split the data into two parts because then I might have different embeddings and it may not preserve the overall structure of the entire dataset.

`numpy==1.24.4
umap-learn==0.5.3
pandas==1.5.3
hdbscan==0.8.33`
